docs/resources/modelarts_dataset.md

```diff
@@ -0,0 +1,195 @@
+---
+subcategory: "Beta"
+---
+
+# huaweicloud_modelarts_dataset
+
+Manages ModelArts dataset resource within HuaweiCloud.
+
+## Example Usage
+
+```hcl
+variable "name" {}
+variable "output_obs_path" {}
+variable "input_obs_path" {}
+
+resource "huaweicloud_modelarts_dataset" "test" {
+  name        = var.name
+  type        = 1
+  output_path = var.output_obs_path
+  description = "Terraform Demo"
+
+  data_source {
+    path = var.input_obs_path
+  }
+
+  labels {
+    name = "foo"
+    type = 1
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the resource. If omitted, the
+ provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String) Specifies the name of the dataset. The name consists of 1 to 100 characters,
+ starting with a letter. Only letters, chinese charactors, digits and underscores (_) are allowed.
+
+* `type` - (Required, String, ForceNew) Specifies the type of dataset. The options are as follows:
+  + **0**: Image classification, supported formats: `.jpg`, `.png`, `.jpeg`, `.bmp`.
+  + **1**: Object detection, supported formats: `.jpg`, `.png`, `.jpeg`, `.bmp`.
+  + **3**: Image segmentation, supported formats: `.jpg`, `.png`, `.jpeg`, `.bmp`.
+  + **100**: Text classification, supported formats: `.txt`, `.csv`.
+  + **200**: Sound classification, Supported formats: `.wav`.
+  + **400**: Table type, supported formats: Carbon type.
+  + **600**: Video, supported formats: `.mp4`.
+  + **900**: Free format.
+
+ Changing this parameter will create a new resource.
+
+* `output_path` - (Required, String, ForceNew) Specifies path for storing output files such as labeled files.
+ The path cannot be the same as the import path or subdirectory of the import path.
+ Changing this parameter will create a new resource.
+
+* `data_source` - (Required, List, ForceNew)Specifies the data sources which be used to imported the source data (such as
+ pictures/files/audio, etc.) in this directory and subdirectories to the dataset. Structure is documented below.
+ Changing this parameter will create a new resource.
+
+* `description` - (Optional, String) Specifies the description of dataset. It contains a maximum of 256 characters and
+ cannot contain special characters `^!<>=&"'`.
+
+* `import_labeled_enabled` - (Optional, Bool, ForceNew) Specifies whether to import labeled files.
+ Default value is `true`. Changing this parameter will create a new resource.
+
+* `label_format` - (Optional, String, ForceNew) Specifies the custom format information of labeled files when import
```

Comment:
The type in the schema is `List` and the declaration posistion should be below parameter `schema`.

Reply:
OK